src/main/scala/mesosphere/marathon/api/v2/GroupsResource.scala

```diff
@@ -136,21 +143,29 @@ class GroupsResource @Inject() (
              @DefaultValue("false")@QueryParam("force") force: Boolean,
              @DefaultValue("false")@QueryParam("dryRun") dryRun: Boolean,
              body: Array[Byte],
-             @Context req: HttpServletRequest, @Context resp: HttpServletResponse): Response = {
-    doIfAuthorized(req, resp, UpdateAppOrGroup, id.toRootPath) { implicit identity =>
-      withValid(Json.parse(body).as[GroupUpdate]) { update =>
+             @Context req: HttpServletRequest): Response = doIfAuthenticated(req) { implicit identity =>
+    withValid(Json.parse(body).as[GroupUpdate]) { groupUpdate =>
+      val newVersion = Timestamp.now()
+      val maybeOldGroup = result(groupManager.group(id.toRootPath))
```

Comment:
You load the group out of transaction. You need to do this for the dryRun, but you should move the authorization logic to updateOrCreate.